src/data/actions/assets.js

```diff
@@ -1,31 +1,47 @@
 import * as clientApi from '../api/client';
 import { assetActions } from '../constants/actionTypes';
 
+const isCorrectResponse = (request, lastRequest) => (
+  request.page === lastRequest.page &&
+  request.sort === lastRequest.sort &&
+  request.direction === lastRequest.direction &&
+  request.assetTypes.Audio === lastRequest.assetTypes.Audio &&
+  request.assetTypes.Code === lastRequest.assetTypes.Code &&
+  request.assetTypes.Documents === lastRequest.assetTypes.Documents &&
+  request.assetTypes.Images === lastRequest.assetTypes.Images &&
+  request.assetTypes.OTHER === lastRequest.assetTypes.OTHER
+);
+
 export const requestAssetsSuccess = response => ({
   type: assetActions.REQUEST_ASSETS_SUCCESS,
-  data: response.assets,
+  data: response,
 });
 
 export const assetDeleteFailure = response => ({
   type: assetActions.DELETE_ASSET_FAILURE,
   response,
 });
 
-export const getAssets = (assetsParameters, courseDetails) =>
-  dispatch =>
+export const getAssets = (request, courseDetails) =>
+  (dispatch, getState) =>
```

Comment:
I'm adding in getState as a parameter so that I can access the current Redux store state.